sklearn/utils/tests/test_utils.py

```diff
@@ -760,3 +762,65 @@ def test_safe_assign(array_type):
     _safe_assign(X, values, column_indexer=column_indexer)
 
     assert_allclose_dense_sparse(X, _convert_container(values, array_type))
+
+
+def test_get_column_indices_interchange():
+    """Check _get_column_indices_interchange for edge cases."""
+    pd = pytest.importorskip("pandas", minversion="1.5")
+
+    df = pd.DataFrame([[1, 2, 3], [4, 5, 6]], columns=["a", "b", "c"])
+    df_interchange = df.__dataframe__()
+
+    key_results = [
+        (slice(1, None), [1, 2]),
+        (slice(None, 2), [0, 1]),
+        (slice(1, 2), [1]),
+        (["b", "c"], [1, 2]),
+        (slice("a", "b"), [0, 1]),
+        (slice("a", None), [0, 1, 2]),
+        (slice(None, "a"), [0]),
+        (["c", "a"], [2, 0]),
+        ([], []),
+    ]
+    for key, result in key_results:
+        assert (
+            _get_column_indices_interchange(
+                df_interchange, key, _determine_key_type(key)
+            )
+            == result
+        )
+
+    msg = "A given column is not a column of the dataframe"
+    with pytest.raises(ValueError, match=msg):
+        _get_column_indices_interchange(df_interchange, ["not_a_column"], "str")
+
+
+def test_polars_indexing():
+    """Check _safe_indexing for polars as expected."""
+    pl = pytest.importorskip("polars", minversion="0.18.2")
+    df = pl.DataFrame({"a": [1, 2, 3, 4], "b": [4, 5, 6, 8], "c": [1, 4, 1, 10]})
+
+    from polars.testing import assert_frame_equal
```

Comment:
can we use directly `pl.testing.assert_frame_equal` instead of importing?

Reply:
The testing module is not exposed in the top level namespace:

```python
import polars as pl
assert not hasattr(pl, "testing")
```